File: 1046-last-stone-weight/1046-last-stone-weight.py
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        heap=[]
        for x in stones:
            heapq.heappush(heap,-x)#max_heap(-val)
            
        while len(heap)>1:
            a=-heapq.heappop(heap)
            b=-heapq.heappop(heap)
            if a!=b: #avoid repeats and zeroes
                heapq.heappush(heap,-(a-b))
        return 0 if not heap else -heap[0]
          
            
            
        
        
        

File: 1046-last-stone-weight/1046-last-stone-weight.java
class Solution {
    public int lastStoneWeight(int[] stones) {
        PriorityQueue<Integer>pq=new PriorityQueue<>((a,b)->b-a);//max_heap;
        for(int x:stones)
            pq.add(x);
        while(pq.size()>1){
            int a=pq.poll(),b=pq.poll();
            pq.add(a-b);
        }
        return pq.peek();
    }
}